src/token-provider.js

```diff
@@ -0,0 +1,23 @@
+import doRequest from './doRequest';
+
+export default class TokenProvider {
+  constructor({ url, queryParams, headers } = {}) {
```

Comment:
This default value is redundant.  Let's say the `TokenProvider` is called with no arguments:
```js
// default value
constructor({ url, queryParams, headers } = {}) {
  this.url = url; // undefined
  this.queryParams = queryParams; // undefined
  this.headers = headers; // undefined
}
```
```js
// no default value
constructor({ url, queryParams, headers }) {
  this.url = url; // undefined
  this.queryParams = queryParams; // undefined
  this.headers = headers; // undefined
}
```
The result is exactly the same.

This also needs some error handling, if any of the arguments are missing.

It would be nice to follow the same initialisation pattern we have for the `PushNotifications` class and expose this as a function, so the user doesn't need to remember to use the `new` keyword. The fact it's a constructor is basically an implementation detail.  So:
```js
// token-provider.js

export default function TokenProvider({ url, queryParams, headers }) {
  // error handling
  return new ActualTokenProvider({ url, queryParams, headers })
}

class ActualTokenProvider {
  // ...
}
```